### DISCUSSION TITLE
ODF / ODT Support

### DISCUSSION THREAD

**Discussion by mathieucloutier86** [dnGrep/dnGrep#1019] [Q&A] [answered]
Hi ! 

I might be mistaken, but I took for granted this morning that dnGrep could find text inside some Open document text file. The only reference to this I was able to find dated back from 2013 where dnGrep apparently couldn't do this. 

Is it still impossible to do ? 

Thanks,
Mathieu

**Comment by doug24** [comment]
No, dnGrep cannot search .odt files as a document like it can with .docx files.  I tried to read one with the OpenXML library, but it did not work. 

However, .odt files are zip archives with xml files containing the document text, so you can search in the xml.

1) In dnGrep Options dialog, add `odt` to the list of Archive file types.  You can also add `odt` to the list of types to open the whole archive:
<img src="https://github.com/dnGrep/dnGrep/assets/17227248/1e468849-50b8-4a1c-9b1a-6a98b2c97645" width="680"/>

2) To search, check the `Search in archives` checkbox, and make sure the file pattern includes `*.xml` or `*.*`.
<img src="https://github.com/dnGrep/dnGrep/assets/17227248/a629a982-b390-4ee2-813b-28b9bd5a4198" width="680"/>

3) The search results will be raw xml, not readable text, but will tell you if your search words or pattern exists in the document.

4) You can open the document from dnGrep by Alt-double clicking the results.

**Comment by mathieucloutier86** [comment]
It's too bad, even if it's better than nothing. Is there a way to do this
eventually, or is it completely impossible? As an archivist, it's a bit sad
to think that an open format like ODF doesn't lend itself as well to a tool
like dnGrep as Word or even WordPerfect. The possibility to preview
multiple files is really a strong argument here to easily find content and
to put it in context with the rest of the text.

Thank you for your quick response,
Mathieu

**Comment by doug24** [comment]
It really depends on if I can find a library (like OpenXML) to read the file and extract the text.  I work on this project on my own time, and I do not have the time implement and test every file format.  I have similar requests for epub and mobi formats.

**Comment by mathieucloutier86** [comment]
No problem, I understand. I'll let you know if I come across something.

Thank you for your time and dedication,
Best,
Mathieu

**Comment by doug24** [comment]
@mathieucloutier86 - I have a solution for Open Document files that will come out in the February release. It uses the Apache Tika application which you will need to install separately from dnGrep (I can't include the installation in the dnGrep installer). I will have instructions for how to install Tika and configure dnGrep to use it.

**Reply by mathieucloutier86** [reply]
I'm really excited to hear about this new development ! This is a wonderful news ! I'm looking forward to test the application. Hopefully, I'll be able to have Apache Tika authorized at my workplace, but I'll be using it for my personal files at home for sure and will be glad to report back on the performance. Are you welcoming donations at this time ? I'd like to support your dedicated work on dngrep.

Best,
Mathieu

**Comment by doug24** [comment, marked as answer]
Available now in [release 4.1.27](https://github.com/dnGrep/dnGrep/releases/tag/v4.1.27.0)
